### ISSUE TITLE
How to save a graph

### ISSUE THREAD

**Issue by dhbello** [tensorflow/models#38]
Quick question: How can I freeze/save my fine-tune model? (To use it later on android)

I'm trying to save it during training checkpoints using graph_util.convert_variables_to_constants(), however, it is not clear with the name of the output classification layer in the retrained graph.

[https://github.com/tensorflow/models/blob/master/inception/inception/inception_train.py#L349](url)
```
      if step % 5000 == 0 or (step + 1) == FLAGS.max_steps:
        output_graph_def = graph_util.convert_variables_to_constants(
                sess, sess.graph.as_graph_def(), ????)
```
Thanks in advanced.

**Comment by markb729** [comment]
Any progress on this issue? I had call to implement the same procedure today and am also having trouble identifying the layer names.

**Comment by petewarden** [comment]
Sorry you're hitting problems, the output layer names can be hard to figure out. Typically for Inception models they're "softmax" (and you'll need them to run the model too). @shlens do you know what your training script names them in this case?

**Comment by shlens** [comment]
Sorry for all of the trouble. We keep a track of pointers to all Tensors of
interest in the Python endpoints dict.

For instance, the logits can be found here:
https://github.com/tensorflow/models/blob/master/inception/inception/slim/inception_model.py#L328

And the normalized predictions can be found here:
https://github.com/tensorflow/models/blob/master/inception/inception/slim/inception_model.py#L329

If you run the script and print out endpoints['predictions'].op.name, this
should provide the name of the output layer.

**Comment by girving** [comment]
@shlens @petewarden Is this resolved?

**Comment by kuza55** [comment]
I've been trying to do the same thing so that I can run this model, and I must say it's been confusing.

I ended up figuring out that I needed to pass in 'inception_v3/logits/predictions' as the output_node (which I only found by iterating over the graph and dumping the names).

I also ended up getting stuck for a moment since I was confused by how to deal with the fact that the tensor was of shape [32, 299, 299, 3] and I couldn't figure out what I needed to do to get it to accept a tensor of the shape [1, 299, 299, 3]; in the end it just seems to work? But I have no idea if I'm doing the right thing.
